rust/src/builders/witness_builder.rs

```diff
@@ -337,7 +342,9 @@ impl TransactionWitnessSetBuilder {
     }
 
     pub fn get_redeemer(&self) -> Redeemers {
-        Redeemers(self.redeemers.clone().into_values().collect())
+        Redeemers(self.redeemers.clone().iter().map(|i|{
```

Comment:
I don't think this clone is necessary anymore since `iter()` should give you refs to iterate over which you are cloning in that map anyway. Same with the one above in `get_plutus_datum`.